api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,444 @@
+                nodeInstance.getName(), LocalDateTime.MIN, LocalDateTime.MAX, Duration.ZERO));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnInstanceTypes() {
+        doReturn(instanceTypes).when(mockInstanceOfferManager).getAllowedInstanceTypes(1L, true);
+
+        assertThat(clusterApiService.getAllowedInstanceTypes(1L, true)).isEqualTo(instanceTypes);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnToolInstanceTypes() {
+        doReturn(instanceTypes).when(mockInstanceOfferManager).getAllowedToolInstanceTypes(1L, true);
+
+        assertThat(clusterApiService.getAllowedToolInstanceTypes(1L, true)).isEqualTo(instanceTypes);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnAllowedInstanceAndPriceTypes() {
+        final AllowedInstanceAndPriceTypes allowedInstanceAndPriceTypes =
+                NodeCreatorUtils.getDefaultAllowedInstanceAndPriceTypes();
+        doReturn(allowedInstanceAndPriceTypes).when(mockInstanceOfferManager)
+                .getAllowedInstanceAndPriceTypes(1L, 2L, true);
+
+        assertThat(clusterApiService.getAllowedInstanceAndPriceTypes(1L, 2L, true))
+                .isEqualTo(allowedInstanceAndPriceTypes);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnMasterNodes() {
+        final MasterNode masterNode = NodeCreatorUtils.getMasterNodeWithEmptyNode();
+        final List<MasterNode> masterNodes = Collections.singletonList(masterNode);
+        doReturn(masterNodes).when(mockNodesManager).getMasterNodes();
+
+        assertThat(clusterApiService.getMasterNodes()).isEqualTo(masterNodes);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnNodeDisksForAdmin() {
+        doReturn(nodeDisks).when(mockNodeDiskManager).loadByNodeId(nodeDisk.getNodeId());
+
+        assertThat(clusterApiService.loadNodeDisks(nodeDisk.getNodeId())).hasSize(1).contains(nodeDisk);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnNodeDisksWhenPermissionIsGranted() {
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(nodeDisks).when(mockNodeDiskManager).loadByNodeId(nodeDisk.getNodeId());
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeDisk.getNodeId(), filterPodsRequest);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeDisk.getNodeId());
+        mockUser();
+        mockRun(pipelineRun);
+
+        assertThat(clusterApiService.loadNodeDisks(nodeDisk.getNodeId())).hasSize(1).contains(nodeDisk);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyAccessToNodeDisksWhenPermissionIsNotGranted() {
+        initAclEntity(nodeInstance);
+        doReturn(nodeDisks).when(mockNodeDiskManager).loadByNodeId(nodeDisk.getNodeId());
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeDisk.getNodeId(), filterPodsRequest);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeDisk.getNodeId());
```

Comment:
It does look like we can just replace it with the new method `mockNode`. Some similar cases with `terminateNode` calls in different tests too. You can just add `terminateNode` to `mockNode` and everything seems to be just fine.